Mathlib/Topology/Algebra/Module/Multilinear/Basic.lean

```diff
@@ -299,19 +299,57 @@ theorem _root_.ContinuousLinearMap.compContinuousMultilinearMap_coe (g : M₂ 
   ext m
   rfl
 
+/-- `ContinuousMultilinearMap.prod` as an `Equiv`. -/
+@[simps apply symm_apply_fst symm_apply_snd, simps (config := .lemmasOnly) symm_apply]
+def prodEquiv :
+    (ContinuousMultilinearMap R M₁ M₂ × ContinuousMultilinearMap R M₁ M₃) ≃
+      ContinuousMultilinearMap R M₁ (M₂ × M₃) where
+  toFun f := f.1.prod f.2
+  invFun f := ((ContinuousLinearMap.fst _ _ _).compContinuousMultilinearMap f,
+    (ContinuousLinearMap.snd _ _ _).compContinuousMultilinearMap f)
+  left_inv _ := rfl
+  right_inv _ := rfl
+
+theorem prod_ext_iff {f g : ContinuousMultilinearMap R M₁ (M₂ × M₃)} :
+    f = g ↔ (ContinuousLinearMap.fst _ _ _).compContinuousMultilinearMap f =
+      (ContinuousLinearMap.fst _ _ _).compContinuousMultilinearMap g ∧
+      (ContinuousLinearMap.snd _ _ _).compContinuousMultilinearMap f =
+      (ContinuousLinearMap.snd _ _ _).compContinuousMultilinearMap g := by
+  rw [← Prod.mk.inj_iff, ← prodEquiv_symm_apply, ← prodEquiv_symm_apply, Equiv.apply_eq_iff_eq]
+
+@[ext]
+theorem prod_ext {f g : ContinuousMultilinearMap R M₁ (M₂ × M₃)}
+    (h₁ : (ContinuousLinearMap.fst _ _ _).compContinuousMultilinearMap f =
+      (ContinuousLinearMap.fst _ _ _).compContinuousMultilinearMap g)
+    (h₂ : (ContinuousLinearMap.snd _ _ _).compContinuousMultilinearMap f =
+      (ContinuousLinearMap.snd _ _ _).compContinuousMultilinearMap g) : f = g :=
+  prod_ext_iff.mpr ⟨h₁, h₂⟩
+
+theorem eq_prod_iff {f : ContinuousMultilinearMap R M₁ (M₂ × M₃)}
+    {g : ContinuousMultilinearMap R M₁ M₂} {h : ContinuousMultilinearMap R M₁ M₃} :
+    f = g.prod h ↔ (ContinuousLinearMap.fst _ _ _).compContinuousMultilinearMap f = g ∧
+      (ContinuousLinearMap.snd _ _ _).compContinuousMultilinearMap f = h :=
+  prod_ext_iff
+
+theorem add_prod_add [ContinuousAdd M₂] [ContinuousAdd M₃]
+    (f₁ f₂ : ContinuousMultilinearMap R M₁ M₂) (g₁ g₂ : ContinuousMultilinearMap R M₁ M₃) :
+    (f₁ + f₂).prod (g₁ + g₂) = f₁.prod g₁ + f₂.prod g₂ :=
+  rfl
```

Comment:
Could you add `smul_prod_smul` too, where the same scalar is used twice?

Reply:
(Ideally with the scalar belonging to an unrelated semiring `S`, but `R` with a TODO comment about generalizing is ok)